src/selectors/search.js

```diff
@@ -0,0 +1,33 @@
+import Fuse from 'fuse.js';
+import { createSelector } from 'reselect';
+
+import { getExternalData } from './protocol';
+
+const getDatasourceKey = (_, props) => props.dataSourceKey;
+const getSearchOpts = (_, props) => props.options;
+
+/**
+ * Look up the search set for a search based on a data source in the protocol.
+ * The value of this key should point to an attribute in the protocol's externalData.
+ */
+export const getSearchData = createSelector(
```

Comment:
Should be factory method:

```js
makeGetSearchData = () =>
   createSelector(
      ...
```

Reply:
OK. Can you explain why this one needs the factory? The underlying memoize & recalculate seems to be working as I'd expect.

Reply:
Nevermind; I see based on your above comments. Thanks!